clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy

```diff
@@ -30,43 +43,44 @@ class AzureResourceManagerClient extends AzureBaseClient {
 
-  public AzureResourceManagerClient(String subscriptionId) {
+  AzureResourceManagerClient(String subscriptionId) {
     super(subscriptionId)
   }
 
-  public String createLoadBalancer(AzureCredentials creds,
-                                   String resourceGroupName,
-                                   String loadBalancerName,
-                                   String region) {
-    String templateVersion = "1.0.0.0"
-
-    try {
-      if (!resourceGroupExists(creds, resourceGroupName)) {
-        createResouceGroup(creds, resourceGroupName, region)
-      }
+  String createLoadBalancerFromTemplate(AzureCredentials credentials,
+                                               String template,
+                                               String resourceGroupName,
+                                               String region,
+                                               String loadBalancerName) {
+    Map<String, String> parameters = new HashMap<String, String>();
+    parameters.put("location", region);
+    createLoadBalancerFromTemplate(credentials, template, parameters, resourceGroupName, region, loadBalancerName)
+  }
 
-      Map<String, String> templateParams = new HashMap<String, String>()
-      templateParams.put("location", region)
-      templateParams.put("addressPrefix", "10.0.0.0/16")
-      templateParams.put("subnetPrefix", "10.0.0.0/24")
-
-      DeploymentExtended deployment = ResourceHelper.createTemplateDeploymentFromURI(
-        this.getResourceManagementClient(creds),
-        resourceGroupName,
-        DeploymentMode.Incremental,
-        loadBalancerName,
-        loadBalancerTemplateUri,
-        templateVersion,
-        templateParams
-      )
-
-      return deployment.name
-    } catch (e) {
-      throw new RuntimeException("Unable to create load balancer ${loadBalancerName}", e)
+  String createLoadBalancerFromTemplate(AzureCredentials credentials,
+                                               String template,
+                                               Map<String, String> templateParams,
+                                               String resourceGroupName,
+                                               String region,
+                                               String loadBalancerName) {
+    if (!resourceGroupExists(credentials, resourceGroupName)) {
+      createResouceGroup(credentials, resourceGroupName, region)
     }
+
+    String deploymentName = loadBalancerName + "_deployment"
+
+    DeploymentExtended deployment = createTemplateDeploymentFromPath(this.getResourceManagementClient(credentials),
+                                                                     resourceGroupName,
+								     DeploymentMode.Incremental,
+								     deploymentName,
+								     template,
+								     templateParams)
+
+    deployment.properties.provisioningState
```

Comment:
returning the deployment object is the right thing but at the time the PR was created not all the functionality relying on it (task monitoring for instance) was available; I've added that now.